records_mover/url/base.py

```diff
@@ -1,4 +1,5 @@
 from contextlib import contextmanager
```

Comment:
There was a lot of common code from the S3 implementation that made sense to be generic here.